api-put-object.go

```diff
@@ -27,78 +27,94 @@ import (
 	"strings"
 )
 
+// toInt - converts go value to its integer representation based
+// on the value kind if it is an integer.
+func toInt(value reflect.Value) (size int64) {
```

Comment:
You can directly do int() on reflect.ValueOf and let the compiler do the type validation.